train.lua

```diff
@@ -131,6 +147,71 @@ local function loadModel(opt, dicts)
 
   checkpoint, opt, paramChanges = onmt.train.Saver.loadCheckpoint(opt)
 
+  if checkpoint.dicts ~= dicts and opt.update_vocab then
```

Comment:
`checkpoint.dicts ~= dicts` is always `true` as you are comparing pointers not content. I think it is reasonable to always execute this code when `opt.update_vocab == true` so just drop the first condition.